File: android/app/src/main/java/edu/gatech/cs2340/rajsjarfiles/spacetrader/entity/game/Game.java
package edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.game;

import edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.universe.Planet;
import edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.universe.Universe;

/**
 * Represents a model of the game.
 */
public class Game {
    private final GameDifficulty gameDiff;

    private final Universe universe;

    /**
     * One arg constructor for creating a game.
     *
     * @param gameDiff the game difficulty
     */
    public Game(GameDifficulty gameDiff) {
        this.gameDiff = gameDiff;
        universe = new Universe();
    }

    /**
     * @return the game difficulty
     */
    public GameDifficulty getGameDiff() {
        return gameDiff;
    }

    /**
     * @return the universe
     */
    public Universe getUniverse() {
        return universe;
    }

    @Override
    public String toString() {
        return "The game difficulty is set to " + gameDiff.toString()
                + ".\n" + universe.toString();
    }

}

File: ios/SpaceTrader/SpaceTrader/src/entity/game/Game.swift
//
//  .swift
//  SpaceTrader
//
//  Created by Thomas Suarez on 4/20/19.
//  Copyright © 2019 RajsJarFiles. All rights reserved.
//

import Foundation

class Game: CustomStringConvertible {
    
    private var gameDiff : GameDifficulty
    private var universe : Universe
    
    init(gameDiff : GameDifficulty) {
        self.gameDiff = gameDiff;
        self.universe = Universe()
    }
    
    func getGameDiff() -> GameDifficulty {
        return self.gameDiff;
    }
    
    func getUniverse() -> Universe {
        return self.universe;
    }
    
    public var description: String {
        return "The game difficulty is set to " + getGameDiff()
    }
    
}
